### ISSUE TITLE
Syntethized speech is too slow / too fast while sample rate is the same as in DS

### ISSUE THREAD

**Issue by vcjob** [mozilla/TTS#288]
Hello everyone!

I trained the Tacotron model and now want to test it. When I listen to the speech from tensorboard it sounds fine. But when it comes to do inference myself through synthesize.py or JupyterNotebook, resulting speech becomes too slow. I attached few examples.
I tried to change sample rate to lower and upper values - the only thing changes is pitch, while speed remains the same. Tried to change some parameters in config, in utils/syntesize.py and even in audio.py - no success.
Please help!
Thank you!

___
Thats what I see when test it:
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > num_mels:80
 | > min_level_db:-100
 | > frame_shift_ms:12.5
 | > frame_length_ms:50
 | > ref_level_db:20
 | > num_freq:1025
 | > power:1.5
 | > preemphasis:0.98
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:10500.0
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > n_fft:2048
 | > hop_length:275
 | > win_length:1100

So, that's obviously the parameters from config.json, which I used to train the model. 

[samle_res.zip](https://github.com/mozilla/TTS/files/3647717/samle_res.zip)

**Comment by erogol** [comment]
which branch?

**Comment by vcjob** [comment]
> which branch?

master

**Comment by erogol** [comment]
try dev. If you use gradual training, don't forget to update "r" for inference. It is done automatically in dev.

**Comment by vcjob** [comment]
> try dev. If you use gradual training, don't forget to update "r" for inference. It is done automatically in dev.

I will now, thank you. 
Regarding the 'r' - I tried to change and none except for 7 works, with other values model fails (which is strange, as it supposed to be 3 by 80000 steps)

**Comment by vcjob** [comment]
> try dev. If you use gradual training, don't forget to update "r" for inference. It is done automatically in dev.

I've just tried dev branch (git checkout dev; python3 setup.py install) and then again python3 synthesize.py text config model output
Results are the same, speech still is too slow.

**Comment by vcjob** [comment]
> All works form me on dev branch so without any more info I cannot help you on that.

Which information can possible help to figure it out?
I will attach down below my full config, also will attach code of the _synthesize.py_ and _train.py_ just in case.
It is strange, while training the model, on the testing stage it does everything the same what in _synthesize.py_ is performed. Both use _from TTS.utils.synthesis import synthesis_. What can be wrong? 
Thank you for any suggestions! 
P.S. changing _"github_branch":"* master",_ to _"github_branch":"* dev",_ didn't do the trick.

[code&wavs.zip](https://github.com/mozilla/TTS/files/3656317/code.wavs.zip)

**Comment by vcjob** [comment]
> All works form me on dev branch so without any more info I cannot help you on that.

I rested the repository, updated all up to date, and now it does not slow down anymore the voice, but speeds it up quite a bit. In my dataset I have same speed everywhere in samples, while output sample is around 1.5 speed of the training one. What can cause that?

One more thing. After update I had to add following line to the synthesize.py: 
    _if ap_vocoder != None:
        postnet_output = ap_vocoder._normalize(postnet_output)_
since it is equal to None when we use no external vocoder, and later on it cause reasonable error (None object has no attribute '_normalize')
